### ISSUE TITLE
feature request: pkey -text or -text_pub should show dsa key size

### ISSUE THREAD

**Issue by bjmgeek** [openssl/openssl#9927]
Here's an example using an RSA key:
````
$ openssl pkey -text_pub -pubout  < rsa.key
-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAuH+lq6jdIttSR/z9pH+o
0MdxkHZPstO30LnZ9LMyxtqOfwj4JApFwbjxjOjqHY0C3WorMVYn2fcx8y7CdcIN
bXaQrl5ywMtLMKj+8qj7Iubc5Aiaxr79kCGgBeRbHvG5YonCj8Xmjs4ZQnYMc9d3
OiPgp8leJ7Voocqk2PS3eCXvLPUNCjKjLt4+M7bqWVg2Pu81cA6gSO0LRvN+2Grs
qOpAN5Ch0b6HW/1RMktXoLj3tsh2G4xwgAT3KgWeOrTmGpmma3QSQx1AKKh4MIEH
WEK3K8DVw2CjtjLP8eXHzacYCsCxJzth1FTZK2tamq/d5BdR0yerpGNUb/ToSA+J
uQIDAQAB
-----END PUBLIC KEY-----
RSA Public-Key: (2048 bit)
Modulus:
    00:b8:7f:a5:ab:a8:dd:22:db:52:47:fc:fd:a4:7f:
    a8:d0:c7:71:90:76:4f:b2:d3:b7:d0:b9:d9:f4:b3:
    32:c6:da:8e:7f:08:f8:24:0a:45:c1:b8:f1:8c:e8:
    ea:1d:8d:02:dd:6a:2b:31:56:27:d9:f7:31:f3:2e:
    c2:75:c2:0d:6d:76:90:ae:5e:72:c0:cb:4b:30:a8:
    fe:f2:a8:fb:22:e6:dc:e4:08:9a:c6:be:fd:90:21:
    a0:05:e4:5b:1e:f1:b9:62:89:c2:8f:c5:e6:8e:ce:
    19:42:76:0c:73:d7:77:3a:23:e0:a7:c9:5e:27:b5:
    68:a1:ca:a4:d8:f4:b7:78:25:ef:2c:f5:0d:0a:32:
    a3:2e:de:3e:33:b6:ea:59:58:36:3e:ef:35:70:0e:
    a0:48:ed:0b:46:f3:7e:d8:6a:ec:a8:ea:40:37:90:
    a1:d1:be:87:5b:fd:51:32:4b:57:a0:b8:f7:b6:c8:
    76:1b:8c:70:80:04:f7:2a:05:9e:3a:b4:e6:1a:99:
    a6:6b:74:12:43:1d:40:28:a8:78:30:81:07:58:42:
    b7:2b:c0:d5:c3:60:a3:b6:32:cf:f1:e5:c7:cd:a7:
    18:0a:c0:b1:27:3b:61:d4:54:d9:2b:6b:5a:9a:af:
    dd:e4:17:51:d3:27:ab:a4:63:54:6f:f4:e8:48:0f:
    89:b9
Exponent: 65537 (0x10001)
````

The key size is listed (2048 bits).  

Here's an example with a dsa key.  I realize that the dsa key includes the parameters, but I'd have to dig to find out the size.
````
-----BEGIN PUBLIC KEY-----
MIIDRzCCAjkGByqGSM44BAEwggIsAoIBAQCN2u6QDNt2WuxdSzgAMM/6fXODDNPa
lvSjIzaA484ODlWK1e0Yb8ZxJl5ILmk33jj/6wpWM+eEKc2lwBsV+7YIUgbHeNR+
hOct0FzCayQp6Eq0cOLUdCo4HxCvScgOBD7+kZ5UqVtgwG7k/S7gtvdcrB7KKSLH
SG/oy2JD0Uh339MAnJHkK3U1iy7g9+4tuXIy+/pu/kpcrPDMjiIVAvRRFjbM6SEW
9+8CLH0xkz8YZJSJ44exPcAhZy2ZacaCKsUdYvzpsKOgwK1Dl+6uJOviVJ/tEBlj
u1ynZ4FpIzK0xKmMTzAI0p1rtc5FpqdQ+PGFpF0gzfodEfT7j+/EGyotAiEAuThq
m8KO5Yh3DptgRysSwG7jvI8wbaPc/TQODMSfsaUCggEANd0cpKPP7rGy4HA9goXq
MeQtQMo3DMtWx5FzRKqf3nAb0RVr5D51YOwKD2h2M7N3vz/5DERq9XwImXqrp0+P
o376FrUKeBhEDS1786NOrLcMLUREDoo98+U4Y5QPubk6rKmoDV4g/eeqze9ilZtY
pGK7BxFpxjFJR79WZX2WeioSuJh7p+4S9zVQQpMH2CoDvfWItcXjLQ1yRHw2yvBW
qsZoaupdjFjindSLmlIZDNJYWWhWn/pWRKPuhbXXRo26SvI4nEh6MaE2zHr4jX7i
pnleCGWUn5cGlMOaLgFwqTYiYPfJme6+lw2vEJ0aAaLmLQVhuTAk0q97nsVw/pdc
gQOCAQYAAoIBAQCJPxcNnHU9xNV215tGT/3m7dXtNypMo5B9kWH8y2l4qqCCYWUC
PbS1OyfkKP1sRFLSwcJYc9tS/td+8ykD5Odf0UC9YBthm8PiHu8CbYxNP0pSCGjL
KMkBLu8fuLbyB+mpfLVT1KWA5UpxDFT6Zn5PFlkj8dpUWgRFfr4HNYZCQueZA1EZ
8y7XjUeIEQWVLY7jmCR7LAwhWGLpy6gU/WjsqRcEJDhTBxWBcsitJkGEMEcsICBk
OJCxzXjJGr1JR/+GSr031DsSDvO8vpHlwP7uOrZG6i/tFHyWqO6AiLyzNn6Smvkf
Iz2u4WWoOGm2tFosXMcjTpeAutVp4TWMonQp
-----END PUBLIC KEY-----
pub: 
    00:89:3f:17:0d:9c:75:3d:c4:d5:76:d7:9b:46:4f:
    fd:e6:ed:d5:ed:37:2a:4c:a3:90:7d:91:61:fc:cb:
    69:78:aa:a0:82:61:65:02:3d:b4:b5:3b:27:e4:28:
    fd:6c:44:52:d2:c1:c2:58:73:db:52:fe:d7:7e:f3:
    29:03:e4:e7:5f:d1:40:bd:60:1b:61:9b:c3:e2:1e:
    ef:02:6d:8c:4d:3f:4a:52:08:68:cb:28:c9:01:2e:
    ef:1f:b8:b6:f2:07:e9:a9:7c:b5:53:d4:a5:80:e5:
    4a:71:0c:54:fa:66:7e:4f:16:59:23:f1:da:54:5a:
    04:45:7e:be:07:35:86:42:42:e7:99:03:51:19:f3:
    2e:d7:8d:47:88:11:05:95:2d:8e:e3:98:24:7b:2c:
    0c:21:58:62:e9:cb:a8:14:fd:68:ec:a9:17:04:24:
    38:53:07:15:81:72:c8:ad:26:41:84:30:47:2c:20:
    20:64:38:90:b1:cd:78:c9:1a:bd:49:47:ff:86:4a:
    bd:37:d4:3b:12:0e:f3:bc:be:91:e5:c0:fe:ee:3a:
    b6:46:ea:2f:ed:14:7c:96:a8:ee:80:88:bc:b3:36:
    7e:92:9a:f9:1f:23:3d:ae:e1:65:a8:38:69:b6:b4:
    5a:2c:5c:c7:23:4e:97:80:ba:d5:69:e1:35:8c:a2:
    74:29
P:   
    00:8d:da:ee:90:0c:db:76:5a:ec:5d:4b:38:00:30:
    cf:fa:7d:73:83:0c:d3:da:96:f4:a3:23:36:80:e3:
    ce:0e:0e:55:8a:d5:ed:18:6f:c6:71:26:5e:48:2e:
    69:37:de:38:ff:eb:0a:56:33:e7:84:29:cd:a5:c0:
    1b:15:fb:b6:08:52:06:c7:78:d4:7e:84:e7:2d:d0:
    5c:c2:6b:24:29:e8:4a:b4:70:e2:d4:74:2a:38:1f:
    10:af:49:c8:0e:04:3e:fe:91:9e:54:a9:5b:60:c0:
    6e:e4:fd:2e:e0:b6:f7:5c:ac:1e:ca:29:22:c7:48:
    6f:e8:cb:62:43:d1:48:77:df:d3:00:9c:91:e4:2b:
    75:35:8b:2e:e0:f7:ee:2d:b9:72:32:fb:fa:6e:fe:
    4a:5c:ac:f0:cc:8e:22:15:02:f4:51:16:36:cc:e9:
    21:16:f7:ef:02:2c:7d:31:93:3f:18:64:94:89:e3:
    87:b1:3d:c0:21:67:2d:99:69:c6:82:2a:c5:1d:62:
    fc:e9:b0:a3:a0:c0:ad:43:97:ee:ae:24:eb:e2:54:
    9f:ed:10:19:63:bb:5c:a7:67:81:69:23:32:b4:c4:
    a9:8c:4f:30:08:d2:9d:6b:b5:ce:45:a6:a7:50:f8:
    f1:85:a4:5d:20:cd:fa:1d:11:f4:fb:8f:ef:c4:1b:
    2a:2d
Q:   
    00:b9:38:6a:9b:c2:8e:e5:88:77:0e:9b:60:47:2b:
    12:c0:6e:e3:bc:8f:30:6d:a3:dc:fd:34:0e:0c:c4:
    9f:b1:a5
G:   
    35:dd:1c:a4:a3:cf:ee:b1:b2:e0:70:3d:82:85:ea:
    31:e4:2d:40:ca:37:0c:cb:56:c7:91:73:44:aa:9f:
    de:70:1b:d1:15:6b:e4:3e:75:60:ec:0a:0f:68:76:
    33:b3:77:bf:3f:f9:0c:44:6a:f5:7c:08:99:7a:ab:
    a7:4f:8f:a3:7e:fa:16:b5:0a:78:18:44:0d:2d:7b:
    f3:a3:4e:ac:b7:0c:2d:44:44:0e:8a:3d:f3:e5:38:
    63:94:0f:b9:b9:3a:ac:a9:a8:0d:5e:20:fd:e7:aa:
    cd:ef:62:95:9b:58:a4:62:bb:07:11:69:c6:31:49:
    47:bf:56:65:7d:96:7a:2a:12:b8:98:7b:a7:ee:12:
    f7:35:50:42:93:07:d8:2a:03:bd:f5:88:b5:c5:e3:
    2d:0d:72:44:7c:36:ca:f0:56:aa:c6:68:6a:ea:5d:
    8c:58:e2:9d:d4:8b:9a:52:19:0c:d2:58:59:68:56:
    9f:fa:56:44:a3:ee:85:b5:d7:46:8d:ba:4a:f2:38:
    9c:48:7a:31:a1:36:cc:7a:f8:8d:7e:e2:a6:79:5e:
    08:65:94:9f:97:06:94:c3:9a:2e:01:70:a9:36:22:
    60:f7:c9:99:ee:be:97:0d:af:10:9d:1a:01:a2:e6:
    2d:05:61:b9:30:24:d2:af:7b:9e:c5:70:fe:97:5c:
    81
````

**Comment by beldmit** [comment]
Should we print the result of `DSA_bits` or `DSA_security_bits`?
BTW, it's a good first issue anyway.

**Comment by jayaram-intel** [comment]
Hi
Submitted changes for review  at #9983 
Please Review the changes.

**Comment by beldmit** [comment]
Fixed via aecf529
